src/sys/windows.rs

```diff
@@ -386,6 +386,64 @@ impl Socket {
         }
     }
 
+    pub fn recv_vectored(&self, bufs: &mut [IoSliceMut<'_>], flags: c_int) -> io::Result<usize> {
+        unsafe {
```

Comment:
Nit: I'm not the biggest fan of these large unsafe blocks, makes it too easy to something unsafe on accident you didn't mean to.

Reply:
I copied the style from the surrounding functions, but I do agree. I reduced the scope of the unsafe blocks in the new functions.